src/layers/base/PixiLayer.ts

```diff
@@ -19,6 +23,7 @@ export abstract class PixiLayer extends Layer {
       this.updateStyle();
 
       const { elm, height, width } = event;
+      const { pixiApplicationOptions } = this.options as PixiLayerOptions;
```

Comment:
I see that the `Layer` handles the possible undefined options from the constructor, by giving it default values.
LGTM 🚢

Reply:
Yes, it guarantees that this.options is a object. But it does not set pixiApplicationOptions. So if pixiApplicationOptions is not passed in to constructor it will be undefined here. But object spread on undefined is allowed https://github.com/tc39/proposal-object-rest-spread/blob/main/Spread.md.

Reply:
Aha, that case only concerns spreading into object literals. 
The [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) itself requires the object to be iterable, which `null` and `undefined` are not.
E.g. `{ ...null, ...undefined }` works, while `[...null, ...undefined]` and `console.log(...undefined)` does not. 🕵🏻‍♂️

Reply:
yes, but that's what we are doing, right?

Reply:
Yup, its all good. Just wasn't aware of this little nuance.
/resolving